styx-common/src/test/java/com/spotify/styx/state/RunStateTest.java

```diff
@@ -95,7 +97,7 @@ public void testTimeTriggerAndRetry() throws Exception {
   public void testTimeTriggerAndRetry2() throws Exception {
     transitioner.initialize(RunState.fresh(WORKFLOW_INSTANCE));
     transitioner.receive(eventFactory.triggerExecution(UNKNOWN_TRIGGER));
-    transitioner.receive(eventFactory.dequeue());
+    transitioner.receive(eventFactory.dequeue(null));
```

Comment:
empty set instead?

Reply:
I intended to leave this test as close to unchanged as possible, so that it tests the same bits of functionality that it tested before (i.e. cover functionality of legacy events, which don't record the set at all). The new tests added below use sets, never `null`.

Reply:
I think we should keep this test class up to date with transition logic, and limit backwards compatible tests to `ReplayEventsTest` class. Current transition logic passes empty sets from `Scheduler.java`, not null. Note that by using Optional to begin with in `EventVisitor` we would get rid of `null` everywhere.